javascript/extractor/src/com/semmle/ts/extractor/TypeScriptParser.java

```diff
@@ -409,6 +409,7 @@ private RuntimeException getExceptionFromMalformedResponse(String response, Exce
         exitCode = parserWrapperProcess.waitFor();
       }
       if (exitCode != null && (exitCode == NODEJS_EXIT_CODE_FATAL_ERROR || exitCode == NODEJS_EXIT_CODE_SIG_ABORT)) {
+        // this is caught in the auto-builder, and handled as an OOM. Check there is the message is changed.
         return new ResourceError("The TypeScript parser wrapper crashed, possibly from running out of memory.", e);
```

Comment:
Let's create a subclass of `ResourceError` and catch that in the auto-builder instead of using string search on the message.